### PR TITLE
Added documentation about enabling OpenTelemetry on Strimzi components

### PR DESCRIPTION
**Pull request by ppatierno**
### Type of change

- Documentation

### Description

This PR adds documentation for #7366 

### Checklist

- [x] Update documentation

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

**Review by im-konge — COMMENTED** [review]
Should we split OpenTracing and OpenTelemetry example configurations by some header or something? Just for better readability. Otherwise LGTM :)

**Comment by ppatierno** [comment]
@im-konge I see your point. This was the result of an offline discussion with @PaulRMellor but let's see what he thinks. I guess he has some more feedback I have to address :-D

**Comment by ppatierno** [comment]
@PaulRMellor any idea why my change could brake the doc build? If you see I have got an error like this ...

```shell
asciidoctor -v --failure-level WARN -t -dbook -a ProductVersion=latest -a BridgeVersion=0.22.1 -a GithubVersion=main -a OAuthVersion=0.10.0 documentation/deploying/deploying.adoc -o documentation/html/deploying.html
asciidoctor: WARNING: assemblies/upgrading/assembly-upgrade-cluster-operator.adoc: line 20: unterminated open block
Input file: documentation/deploying/deploying.adoc
  Time to read and parse source: 0.10582
  Time to convert document: 0.05955
  Total time (read, parse and convert): 0.16537
make: *** [Makefile:119: docu_html] Error 1
```

**Comment by PaulRMellor** [comment]
> @PaulRMellor any idea why my change could brake the doc build? If you see I have got an error like this ...
> 
> ```shell
> asciidoctor -v --failure-level WARN -t -dbook -a ProductVersion=latest -a BridgeVersion=0.22.1 -a GithubVersion=main -a OAuthVersion=0.10.0 documentation/deploying/deploying.adoc -o documentation/html/deploying.html
> asciidoctor: WARNING: assemblies/upgrading/assembly-upgrade-cluster-operator.adoc: line 20: unterminated open block
> Input file: documentation/deploying/deploying.adoc
>   Time to read and parse source: 0.10582
>   Time to convert document: 0.05955
>   Total time (read, parse and convert): 0.16537
> make: *** [Makefile:119: docu_html] Error 1
> ```

The `--` on line 28 might be causing the break. You removed the closing one.

**Comment by PaulRMellor** [comment]
> @im-konge I see your point. This was the result of an offline discussion with @PaulRMellor but let's see what he thinks. I guess he has some more feedback I have to address :-D

@im-konge, @ppatierno - the best we can do within a procedure in terms of headings is using `.` to create an inline heading, which we have for each of examples. We could group under a `==` type heading, but then we would have to take them out of the procedure. Looking at the preview, it looks fine as it is, and will look better when rendered. So I think we should leave it as it is, particularly when opentracing is due to be dropped.

**Review by PaulRMellor — APPROVED** [review]
Nice! :+1:

File: documentation/modules/tracing/proc-enabling-tracing-in-connect-mirror-maker-bridge-resources.adoc [hunk 1]
```diff
@@ -5,22 +5,28 @@
...
-. In the `spec.template` property, configure the Jaeger tracer service. For example:
+. In the `spec.template` property, configure the tracer service.
+
+* Use the xref:ref-tracing-environment-variables-{context}[tracing environment variables] as template configuration properties.
+* For OpenTracing, set the `spec.tracing.type` property to `jaeger`.
+* For OpenTelemetry, set the `spec.tracing.type` property to `opentelemetry`.
 +
 --
-.Jaeger tracer configuration for Kafka Connect
+.Example tracing configuration for Kafka Connect using OpenTracing
```
[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/modules/tracing/proc-enabling-tracing-in-connect-mirror-maker-bridge-resources.adoc:29]: Yes you are right! I will change the order.

**Review by im-konge — APPROVED** [review]

**Pull request merged**